main/mafia.hs

```diff
@@ -304,15 +311,41 @@ pConstraint =
 
 pScriptPath :: Parser File
 pScriptPath =
-  argument textRead $
-       metavar "SCRIPT"
-    <> help "The path to the Haskell shell script."
+  argument scriptRead $
+       metavar "SCRIPT_PATH"
+    <> help "The path to a Haskell script to execute."
 
 pScriptArgs :: Parser File
 pScriptArgs =
   argument textRead $
        metavar "SCRIPT_ARGUMENTS"
-    <> help "Argument to pass to the shell script."
+    <> help "Arguments to pass to the script."
+
+scriptRead :: ReadM File
+scriptRead =
+  eitherReader $ \path ->
+    if isScriptPath path then
+      Right $ T.pack path
```

Comment:
This can't be triggered at this stage right? Should it be `readerError` to make that clear?